uwblueprint/project-read-frontend
# Add a class dialog

### Notion ticket link



[Add a class dialog](https://www.notion.so/uwblueprintexecs/Developer-Hub-30e9b6c82af24054b31acf7c5e822c89?p=ebf3904ee09743cdaa888f61f20fe37e)

WAIT FOR [BACKEND](https://github.com/uwblueprint/project-read-backend/pull/125) TO GO IN FIRST!



### Implementation description

- Added an "add a class" dialog to the sessions page, which can be accessed via the '+' button tab
- This opens a modal prompt that allows the admin user to add a class into the session
<img width="950" alt="Screen Shot 2021-09-08 at 10 55 00 PM" src="https://user-images.githubusercontent.com/47638847/132614719-5916e556-769d-452c-ad26-166ec8ace360.png">
- Upon submit, this dialog calls the POST /classes endpoint and creates the class



### Steps to test

1. Start backend and frontend, and manually test the dialog with a new class. Make sure that the class shows up in http://localhost:8000/sessions/{session_id}/



### What should reviewers focus on?

- React convention, correct modularization & placement of code

### Checklist
